src/datanode/src/region_server.rs

```diff
@@ -96,6 +96,17 @@ impl RegionServer {
             .collect()
     }
 
+    pub fn set_writable(&self, region_id: RegionId, writable: bool) -> Result<()> {
+        let engine = self
+            .inner
+            .region_map
+            .get(&region_id)
+            .with_context(|| RegionNotFoundSnafu { region_id })?;
```

Comment:
```suggestion
            .context(RegionNotFoundSnafu { region_id })?;
```

Reply:
Error with extra fields can run slightly faster if using `with_context`. It can avoid constructing the error every time